techmatters/webchat
# Refactor configuration loading in webchat to facilitate automation

## Description
This PR:
- Adds a pre-build script to webpack which will copy the specified config file (via `CONFIG` env var) into `/src/config.ts` before building. Will throw an error without building if the specified file does not exists.
- Adapts the code in `/src` to work with the above, and dynamically `require`s the config file. Will throw an error if for some reason the file is missing.
- Modifies `custom_helpline.yml` deploy action to pass `CONFIG=${{ steps.helpline_code.outputs.lowercase }}-${{ inputs.environment }}` env var to the build step (e.g. `as-development`).
- Renames the files under `/configurations/*.ts` where necessary, to make them match the above pattern.

### Checklist
- [x] [Corresponding issue has been opened](https://tech-matters.atlassian.net/jira/software/c/projects/CHI/boards/7?modal=detail&selectedIssue=CHI-1564)
- [n/a?] New tests added

### Verification steps
- Locally running `CONFIG=xx-yyyy npm run build` should succeed, for any valid config specification `xx-yyyy`.
  - Confirm that only the account sid and flow sid present in the target config are present in the `/build/bundle.js` file.
- Deployments from this branch should succeed (as QA should probably deploy to some/all staging accounts).
